lessons/durablefunctions/chaining-ts.md

```diff
@@ -0,0 +1,631 @@
+
+### 0.3 Mechanics of Durable Functions
+
+Durable Functions consist of three components:
+
+* Orchestrator Functions: the central part of the Durable framework that orchestrates the actions that should take place by triggering Activity Functions.
+* Activity Functions: the basic workers that execute the single tasks scheduled via the Orchestrator Function.
+* Client Function: the gateway to the Orchestrator Function. The Client Function triggers the Orchestrator Function and serves as the single point of entry for requests from the caller like getting the status of the processing, terminating the processing etc.
+
+Let us assume the following simple execution sequence with two tasks triggered by an HTTP request:
+
+```typescript
+let x = await ctx.CallActivityAsync("F1")
+let y = await ctx.CallActivityAsync("F2", x)
+```
+
+The second task depends on the result of the first task.
+
+The schematic setup with Azure Durable Functions looks like this:
+
+![Durable Function Execution Schema](../../img/lessons/durablefunctions/chaining/SchemaDurableFunction0.png)
+
+The Client Function is triggered by an HTTP request and consequently triggers the Orchestrator function. Internally this means that a message is enqueued in a task hub. We do not have to care about that as we will se later.
+
+![Durable Function Execution Trigger](../../img/lessons/durablefunctions/chaining/SchemaDurableFunction1.png)
+
+After that the Client Functions scales back down to zero and the Orchestrator Function takes over. It fetches the task from the task hub and schedules the Activity Function.
+
+![Durable Function Execution Orchestrator](../../img/lessons/durablefunctions/chaining/SchemaDurableFunction2.png)
+
+Again the orchestrator scales down to zero and the Activity Function is executed and returns its results.
+
+![Durable Function Execution Activity](../../img/lessons/durablefunctions/chaining/SchemaDurableFunction3.png)
+
+After the execution of the first Activity Function the Orchestrator function is invoked again and checks if there are tasks left to do. In our scenario the second Activity Functions is executed. This cycle continuos until all Activity Function calls implemented in the Orchestrator are executed.
+
+After this more theoretical discussion Let us make our hands dirty with some code. But before let us sort our some prerequisites.
+
+### 0.4 Prerequisites for local development
+
+As we could see Azure Durable Functions need some persistency attached. If you want to do a pure local development you must install the [Microsoft Azure Storage Emulator](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator). Up to now the Durable Functions are not compatible with Azurite. So if you are on a non-Windows machine you must use a hybrid approach and connect your Durable Functions to a storage in Azure.

```

Comment:
Like I mentioned earlier, just a table with the prerequisites per exercise would be enough. There's already a prerequisites.md where you can link to that contains descriptions for these tools. If you think that file needs to be updated feel free to include that in this PR.

Reply:
Done